crates/noirc_evaluator/src/ssa_refactor/ir/instruction.rs

```diff
@@ -200,14 +211,32 @@ impl Instruction {
                         return Remove;
                     }
                 }
-                None
             }
-            Instruction::Truncate { .. } => None,
-            Instruction::Call { .. } => None,
-            Instruction::Allocate { .. } => None,
-            Instruction::Load { .. } => None,
-            Instruction::Store { .. } => None,
+            Instruction::ArrayGet { array, index } => {
+                let array = dfg.get_array_constant(*array);
+                if let (Some(array), Some(index)) = (array, dfg.get_numeric_constant(*index)) {
+                    let index =
+                        index.try_to_u64().expect("Expected array index to fit in u64") as usize;
+                    assert!(index < array.len());
+                    return SimplifiedTo(array[index]);
+                }
+            }
+            Instruction::ArraySet { array, index, value } => {
+                let array = dfg.get_array_constant(*array);
+                if let (Some(array), Some(index)) = (array, dfg.get_numeric_constant(*index)) {
+                    let index =
+                        index.try_to_u64().expect("Expected array index to fit in u64") as usize;
+                    assert!(index < array.len());
+                    return SimplifiedTo(dfg.make_array(array.update(index, *value)));
+                }
+            }
```

Comment:
```suggestion
            Instruction::ArrayGet { array, index } => {
                let array = dfg.get_array_constant(*array);
                if let (Some(array), Some(index)) = (array, dfg.get_numeric_constant(*index)) {
                    let index =
                        index.try_to_u64().expect("Expected array index to fit in u64") as usize;
                    assert!(index < array.len());
                    SimplifiedTo(array[index])
                } else {
                    None
                }
            }
            Instruction::ArraySet { array, index, value } => {
                let array = dfg.get_array_constant(*array);
                if let (Some(array), Some(index)) = (array, dfg.get_numeric_constant(*index)) {
                    let index =
                        index.try_to_u64().expect("Expected array index to fit in u64") as usize;
                    assert!(index < array.len());
                    SimplifiedTo(dfg.make_array(array.update(index, *value)))
                } else {
                    None
                }
            }
```

Just as a general style point here, I think it's clearer to explicitly return `None` in these cases with `if` statements rather than falling out the bottom of the match statement.